frontend/src/test/scala/bloop/bsp/BspClientTest.scala

```diff
@@ -104,12 +106,23 @@ object BspClientTest {
       val runningClientServer = lsServer.startTask.runAsync(scheduler)
 
       val cwd = TestUtil.getBaseFromConfigDir(configDirectory.underlying)
-      val initializeServer = endpoints.Build.initialize.request(
-        InitializeBuildParams(
-          rootUri = cwd.toAbsolutePath.toString,
-          Some(BuildClientCapabilities(List("scala")))
-        )
-      )
+      val initializeServer = {
+        def messages = logger0.underlying.getMessages().filter(_._1 == "info")
+        def waitForLiveServer: Task[Unit] = Task(()).flatMap { _ =>
+          if (!messages.exists(_._2.startsWith("The server started to listen")))
```

Comment:
This may be a personal preference, but I would create a string in BspServer with `The server started to listen` and use it here and when building the message string, to avoid a minor change of this string from breaking anything

Reply:
I'd do the same if this was the main code, but it's just test sources :smile:, so I didn't really put too much thinking into it. I've found that inlining the expected messages in the test sources makes it easier to add more tests and fix them if they regress.